### ISSUE TITLE
Circular loop downloading a dependent module

### ISSUE THREAD

**Issue by davereid** [drush-ops/drush#1489]
I have the url_embed module downloaded to my /modules directory. It lists `- composer_manager` under its dependencies in url_embed.info.yml. This is what happens when I try to enable it:

```bash
davereid@davereidair /var/www/drupal8.dev/modules (8.0.x✗)$ rm -rf composer_manager/
davereid@davereidair /var/www/drupal8.dev/modules (8.0.x✗)$ drush pm-enable url_embed -y -v
Include /Users/davereid/opt                                                                                                                                             [notice]
Initialized Drupal 8.0.0-dev root directory at /Users/davereid/Sync/projects/drupal8                                                                                    [notice]
Initialized Drupal site default at sites/default                                                                                                                        [notice]
Executing: mysql --defaults-extra-file=/private/tmp/drush_0v0S56 --database=drupal8 --host=localhost --silent  < /private/tmp/drush_uGRnAA
Executing: mysql --defaults-extra-file=/private/tmp/drush_w1uJXX --database=drupal8 --host=localhost --silent  < /private/tmp/drush_psCArY
Loading release_info engine.                                                                                                                                            [notice]
Downloading release history from https://updates.drupal.org/release-history/composer_manager/8.x                                                                        [notice]
/Users/davereid/.drush/cache/download/https---updates.drupal.org-release-history-composer_manager-8.x retrieved from cache.                                             [notice]
The following projects have unmet dependencies:                                                                                                                      [ok]
url_embed requires composer_manager
Would you like to download them? (y/n): y
Calling proc_open(/usr/local/php5/bin/php /Users/davereid/Sync/opt/drush/drush.php --php=/usr/local/php5/bin/php  --verbose --root=/var/www/drupal8.dev --uri=http://drupal8.dev/  pm-download composer_manager 2>&1);
Executing: git rev-parse --show-toplevel 2> /dev/null
Include /Users/davereid/opt                                                                                                                                             [notice]
Initialized Drupal 8.0.0-dev root directory at /Users/davereid/Sync/projects/drupal8                                                                                    [notice]
Initialized Drupal site drupal8.dev at sites/default                                                                                                                    [notice]
Executing: mysql --defaults-extra-file=/private/tmp/drush_Js5h7M --database=drupal8 --host=localhost --silent  < /private/tmp/drush_rYIw32
Executing: mysql --defaults-extra-file=/private/tmp/drush_17oh6l --database=drupal8 --host=localhost --silent  < /private/tmp/drush_cxLPvt
Loading version_control engine.                                                                                                                                         [notice]
Loading package_handler engine.                                                                                                                                         [notice]
Executing: wget --version
Executing: curl --version
Loading release_info engine.                                                                                                                                            [notice]
Downloading release history from https://updates.drupal.org/release-history/composer_manager/8.x                                                                        [notice]
/Users/davereid/.drush/cache/download/https---updates.drupal.org-release-history-composer_manager-8.x retrieved from cache.                                             [notice]
Undefined index: recommended_major Project.php:383                                                                                                                      [notice]
Undefined index: recommended_major Project.php:384                                                                                                                      [notice]
Undefined index: recommended_major Project.php:388                                                                                                                      [notice]
Downloading project composer_manager ...                                                                                                                                [notice]
/Users/davereid/.drush/cache/download/http---ftp.drupal.org-files-projects-composer_manager-8.x-1.0-beta3.tar.gz retrieved from cache.                                  [notice]
Downloading composer_manager-8.x-1.0-beta3.tar.gz was successful.                                                                                                       [notice]
Md5 checksum of composer_manager-8.x-1.0-beta3.tar.gz verified.                                                                                                         [notice]
Executing: tar  -C /tmp/drush_tmp_1436795462_55a3c24647e3f -xzf composer_manager-8.x-1.0-beta3.tar.gz
Executing: tar -tzf composer_manager-8.x-1.0-beta3.tar.gz
Using destination directory /Users/davereid/Sync/projects/drupal8//modules                                                                                              [notice]
Executing: svn info /Users/davereid/Sync/projects/drupal8//modules
Executing: bzr root /Users/davereid/Sync/projects/drupal8//modules
Project composer_manager (8.x-1.0-beta3) downloaded to /Users/davereid/Sync/projects/drupal8//modules/composer_manager.                                              [success]
Loading composer repositories with package information
Installing dependencies
Nothing to install or update
Generating autoload files
Command dispatch complete                                                                                                                                               [notice]
The following projects have unmet dependencies:                                                                                                                      [ok]
url_embed requires composer_manager
Would you like to download them? (y/n): y
Calling proc_open(/usr/local/php5/bin/php /Users/davereid/Sync/opt/drush/drush.php --php=/usr/local/php5/bin/php  --verbose --root=/var/www/drupal8.dev --uri=http://drupal8.dev/  pm-download composer_manager 2>&1);
Executing: git rev-parse --show-toplevel 2> /dev/null
Include /Users/davereid/opt                                                                                                                                             [notice]
Initialized Drupal 8.0.0-dev root directory at /Users/davereid/Sync/projects/drupal8                                                                                    [notice]
Initialized Drupal site drupal8.dev at sites/default                                                                                                                    [notice]
Executing: mysql --defaults-extra-file=/private/tmp/drush_ET9Oui --database=drupal8 --host=localhost --silent  < /private/tmp/drush_r9SWrc
Executing: mysql --defaults-extra-file=/private/tmp/drush_HoHxmG --database=drupal8 --host=localhost --silent  < /private/tmp/drush_l9v7aD
Loading version_control engine.                                                                                                                                         [notice]
Loading package_handler engine.                                                                                                                                         [notice]
Executing: wget --version
Executing: curl --version
Loading release_info engine.                                                                                                                                            [notice]
Downloading release history from https://updates.drupal.org/release-history/composer_manager/8.x                                                                        [notice]
/Users/davereid/.drush/cache/download/https---updates.drupal.org-release-history-composer_manager-8.x retrieved from cache.                                             [notice]
Undefined index: recommended_major Project.php:383                                                                                                                      [notice]
Undefined index: recommended_major Project.php:384                                                                                                                      [notice]
Undefined index: recommended_major Project.php:388                                                                                                                      [notice]
Downloading project composer_manager ...                                                                                                                                [notice]
/Users/davereid/.drush/cache/download/http---ftp.drupal.org-files-projects-composer_manager-8.x-1.0-beta3.tar.gz retrieved from cache.                                  [notice]
Downloading composer_manager-8.x-1.0-beta3.tar.gz was successful.                                                                                                       [notice]
Md5 checksum of composer_manager-8.x-1.0-beta3.tar.gz verified.                                                                                                         [notice]
Executing: tar  -C /tmp/drush_tmp_1436795545_55a3c299430f2 -xzf composer_manager-8.x-1.0-beta3.tar.gz
Executing: tar -tzf composer_manager-8.x-1.0-beta3.tar.gz
Using destination directory /Users/davereid/Sync/projects/drupal8//modules                                                                                              [notice]
Executing: svn info /Users/davereid/Sync/projects/drupal8//modules
Executing: bzr root /Users/davereid/Sync/projects/drupal8//modules
Install location /Users/davereid/Sync/projects/drupal8//modules/composer_manager already exists. Do you want to overwrite it? (y/n):
```

I can reproduce this locally with the absolute latest Drush master branch. See https://github.com/LionsAd/drupal_ti/issues/14

**Comment by davereid** [comment]
And this happens with any dependency that is not yet locally available, not just specific to composer_manager.

**Comment by davereid** [comment]
Note that the composer_manager module actually downloaded correctly and is available (verified after quitting) at the time when I'm asked to download it again.

**Comment by jonhattan** [comment]
This is a long standing issue I should apply myself to fix! Dupe of #5
